app/src/main/java/org/oppia/android/app/home/HomeViewModel.kt

```diff
@@ -0,0 +1,179 @@
+   * automatically update if constituent parts of the UI change (e.g. if the promoted story list changes).
+   * */
+  val homeItemViewModelListLiveData: LiveData<List<HomeItemViewModel>> by lazy {
+    Transformations.map(homeItemViewModelListDataProvider.toLiveData()) { itemListResult ->
+      if (itemListResult.isFailure()) {
+        logger.e(
+          "HomeFragment",
+          "Failed to retrieve items for home fragment",
+          itemListResult.getErrorOrNull()
+        )
+      }
+      return@map itemListResult.getOrDefault(listOf())
+    }
+  }
+
+  /**
+   * Returns a [HomeItemViewModel] corresponding to the welcome message (see [WelcomeViewModel]), or null if
+   * the specified profile has insufficient information to show the welcome message.
+   * */
+  private fun computeWelcomeViewModel(profile: Profile): HomeItemViewModel? {
+    return if (profile.name.isNotEmpty()) {
+      WelcomeViewModel(fragment, oppiaClock, profile.name)
+    } else null
+  }
+
+  /**
+   * Returns a [HomeItemViewModel] corresponding to the promoted stories to be displayed for this learner
+   * (see [PromotedStoryListViewModel]), or null if this profile does not have any promoted stories.
+   * Promoted stories are determined by any recent stories started by this profile.
+   * */
+  private fun computePromotedStoryListViewModel(
+    ongoingStoryList: OngoingStoryList
+  ): HomeItemViewModel? {
+    val storyViewModelList = computePromotedStoryViewModelList(ongoingStoryList)
+    return if (storyViewModelList.isNotEmpty()) {
+      return PromotedStoryListViewModel(
+        activity,
+        internalProfileId,
+        intentFactoryShim,
+        storyViewModelList
+      )
+    } else null
+  }
+
+  /**
+   * Returns a list of [HomeItemViewModel]s corresponding to the the [PromotedStoryListViewModel] displayed
+   * for this profile (see [PromotedStoryViewModel]), or an empty list if the profile does not have any
+   * ongoing stories at all.
+   */
+  private fun computePromotedStoryViewModelList(
+    ongoingStoryList: OngoingStoryList
+  ): List<PromotedStoryViewModel> {
+    val storyList = if (ongoingStoryList.recentStoryCount != 0) {
+      ongoingStoryList.recentStoryList
+    } else {
+      // TODO(#936): Optimise this as part of recommended stories.
+      ongoingStoryList.olderStoryList
+    }
+    return storyList.take(promotedStoryListLimit)
+      .map { promotedStory ->
+        PromotedStoryViewModel(
+          activity,
+          internalProfileId,
+          intentFactoryShim,
+          storyList.size,
+          storyEntityType,
+          promotedStory
+        )
+      }
+  }
+
+  /**
+   * Returns a list of [HomeItemViewModel]s corresponding to all the lesson topics available and to be
+   * displayed on the home activity (see [TopicSummaryViewModel]) along with associated topics list header (see
+   * [AllTopicsViewModel]). Returns null if there are no lesson topics to display in the home fragment.
```

Comment:
Is the ending sentence still correct per recent changes?

Reply:
Done